### ISSUE TITLE
Consul cluster has no elected leader

### ISSUE THREAD

**Issue by Xuan-Goldenf** [hashicorp/consul#11912]
consul ver 1.11.1
lib: java consul-client https://github.com/rickfast/consul-client
when get healthy services header always get "Consul cluster has no elected leader"

but consul 1.10.0 working fine.

any suggestion?

**Comment by Amier3** [comment]
Hey @Xuan-Goldenf ,

Welcome to the consul community! 

Would love to help with this, but I'll need a bit more information first. Are you able to access the consul cluster directly? If so, could you provide us with: 

- The error logs showing that there's no elected leader ( this should come up in `consul monitor` ) 
- Your consul agent configuration file 

I'll also note that the `consul-client` library isn't supported by us, so we can't guarantee we can help if this issue is related to the library's configuration. If we find that is the case, i'd recommend making an issue in their repo.

**Comment by blake** [comment]
Hi @Xuan-Goldenf, this sounds like it may be related to rickfast/consul-client#457. Can you follow the suggestion in that issue about disabling streaming to see if it resolves your issue?

**Comment by windonis** [comment]
Yeah this issue related for this header ( [X-Consul-KnownLeader](https://developer.hashicorp.com/consul/api-docs/features/blocking#streaming-backend) ) 

That is why we use 1.9.0 for java related clients
